src/transport/GasTransport.cpp

```diff
@@ -362,6 +364,12 @@ void GasTransport::init(thermo_t* thermo, int mode, int log_level)
         }
     }
 
+	//define m_chargeSpecies
+	m_chargeSpecies.resize(m_nsp, 0.0);
+	for (size_t i = 0; i < m_nsp; i++) {
```

Comment:
If you're going to store the species charges, you should store just the charges for non-neutral species, so the loops later can be over that list, rather than having to iterate over the (probably much larger) list of non-charged species every time.